tests/components/home_connect/test_select.py

```diff
@@ -52,6 +57,36 @@ async def test_select(
     assert config_entry.state is ConfigEntryState.LOADED
 
 
+async def test_filter_unknown_programs(
+    bypass_throttle: Generator[None],
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    integration_setup: Callable[[], Awaitable[bool]],
+    setup_credentials: None,
+    get_appliances: Mock,
+    appliance: Mock,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test select that programs that are not part of the official Home Connect API specification are filtered out."""
+    appliance.status.update(SETTINGS_STATUS)
+    appliance.get_programs_available.return_value = [
+        PROGRAM,
+        "NonOfficialProgram",
+        "AntotherNonOfficialProgram",  # We use another program to ensure that programs are iterated over a copy of the list.
```

Comment:
I'm pretty sure this line is too wide for the code base style and should be wrapped or moved up above. (perhaps the docstring too, i can't tell)